libraries/botbuilder-dialogs-adaptive/src/input/choiceInput.ts

```diff
@@ -136,4 +132,16 @@ export class ChoiceInput extends InputDialog {
         return `ChoiceInput[${ this.prompt && this.prompt.toString() }]`;
     }
 
+    private determineCulture(dc: DialogContext, opt: FindChoicesOptions = null): string {
+        const optLocale: string = opt && opt.locale ? opt.locale : null;
+        let culture: string = PromptCultureModels.mapToNearestLanguage(
+            dc.context.activity.locale ||
+            optLocale ||
+            (this.defaultLocale && this.defaultLocale.getValue(dc.state)));
+        if (!culture || !ChoiceInput.defaultChoiceOptions.hasOwnProperty(culture)) {
```

Comment:
Can you add a blank line in between?

Reply:
Do you know why the behavior in "defaultChoices" changed between Prompts and Inputs?

Reply:
I'm not sure what you mean by the behavior in defaultChoices or which changes you're referring to. Are you asking why the inputs have fewer fallbacks or why they didn't use `mapToNearestLanguage` or why they lacked `determineCulture` functions? In any case it's pretty perplexing to me that so much functionality was ever duplicated, since it seems like the adaptive dialog code should have been relying on the existing prompts.